File: src/data/promo-codes.ts
import type { PromoCode } from '@/types';

export const promoCodes: PromoCode[] = [
  {
    code: 'LUNA10',
    type: 'percentage',
    value: 10,
    label: {
      hr: '10% popusta',
      en: '10% off',
    },
  },
  {
    code: 'BEBA20',
    type: 'fixed',
    value: 20,
    minOrderAmount: 50,
    label: {
      hr: '\u20AC20 popusta na narud\u017Ebe iznad \u20AC50',
      en: '\u20AC20 off orders over \u20AC50',
    },
  },
  {
    code: 'FREESHIP',
    type: 'free-shipping',
    value: 0,
    label: {
      hr: 'Besplatna dostava',
      en: 'Free shipping',
    },
  },
];

File: backend/spec/factories/promo_codes.rb
FactoryBot.define do
  factory :promo_code do
    sequence(:code) { |n| "PROMO#{n}" }
    discount_type { :percentage }
    value { 10 }
    label_hr { "10% popusta" }
    label_en { "10% off" }
    active { true }

    trait :fixed do
      discount_type { :fixed }
      value { 2000 }
      min_order_amount { 5000 }
      label_hr { "20€ popusta" }
      label_en { "20€ off" }
    end

    trait :free_shipping do
      discount_type { :free_shipping }
      value { 0 }
      label_hr { "Besplatna dostava" }
      label_en { "Free shipping" }
    end

    trait :expired do
      expires_at { 1.day.ago }
    end

    trait :inactive do
      active { false }
    end

    trait :maxed_out do
      max_uses { 1 }
      current_use_count { 1 }
    end
  end
end
